app/controllers/companies_controller.rb

```diff
@@ -1,6 +1,7 @@
 class CompaniesController < ApplicationController
-  expose :articles, -> { ArticleDecorator.wrap(paginate(authorized_articles)) }
-  expose :company, find_by: :slug
+  expose :article, scope: -> { authorized_articles }
+  expose :articles, -> { ArticleDecorator.wrap(paginate(filtered_articles)) }
+  expose :company, -> { Company.find_by!(slug: request.subdomain) }
```

Comment:
What do you think about creating a `current_company` method in `app/controllers/concerns/authentication.rb` or any other appropriate concern and using `current_company` in all controllers where it's needed?

```ruby
module Authentication
  ...
  included do
    ...
    helper_method :current_user, :current_company
  end
  ...

  def current_company
    @current_company ||= Company.find_by(slug: request.subdomain)
  end
```